homeassistant/components/tibber/sensor.py

```diff
@@ -514,3 +514,89 @@ def get_live_measurement(self):
             _LOGGER.error(errors[0])
             return None
         return self.data.get("data", {}).get("liveMeasurement")
+
+
+class TibberDataCoordinator(update_coordinator.DataUpdateCoordinator):
+    """Handle Tibber data and insert statistics."""
+
+    def __init__(self, hass, tibber_connection):
+        """Initialize the data handler."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=f"Tibber {tibber_connection.name}",
+            update_interval=timedelta(hours=1),
+        )
+        self._tibber_connection = tibber_connection
+
+    async def _async_update_data(self):
+        """Update data via API."""
+        await self._tibber_connection.fetch_consumption_data_active_homes()
+        await self._insert_statistics()
+
+    async def _insert_statistics(self):
+        """Insert Tibber statistics."""
+        for home in self._tibber_connection.get_homes():
+            if not home.hourly_consumption_data:
+                continue
+
+            statistic_id = (
+                f"{TIBBER_DOMAIN}:{home.home_id.replace('-', '')}_energy_consumption"
+            )
+
+            last_stats = await self.hass.async_add_executor_job(
+                get_last_statistics, self.hass, 1, statistic_id, True
+            )
+
+            if not last_stats:
+                hourly_consumption_data = await home.get_historic_data(5 * 365 * 24)
+
+                _sum = 0
+                last_stats_time = None
+            else:
```

Comment:
We should also check the period of the `last_stats`, to handle HA being down for some time for example?

Reply:
It will always update the last 30 days of data.

Reply:
OK, so `home.hourly_consumption_data` holds 30 days worth of data and time `_insert_statistics` is called we'll upsert the past 30 days worth of data?

When we call `statistics_during_period` to get past statistics, we do that for 30 days - 1 hour, and then rewrite history?

Please add comments which explain this.